src/main/java/genesis/Genesis.java

```diff
@@ -0,0 +1,184 @@
+package genesis;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+import utility.ConsolePrinter;
+import utility.FileHandler;
+
+import exception.GenesisException;
+import exception.UnknownCommandException;
+
+public class Genesis {
+
+    private ArrayList<Task> tasks;
+
+    public Genesis() {
+        try {
+            this.tasks = FileHandler.loadFromFile();
+            System.out.println("Task list loaded successfully.");
+        } catch (IOException e) {
+            this.tasks = new ArrayList<>();
+            System.out.println("There is no existing task list found. Initializing new task list.");
+        } catch (IndexOutOfBoundsException | GenesisException e) {
+            this.tasks = new ArrayList<>();
+            System.out.println("Task list is corrupted. Initializing new task list.");
+        } finally {
+            ConsolePrinter.breakLine();
+        }
+    }
+
+    private void handleListTasks() throws GenesisException {
+        if (tasks.size() < 1) {
+            throw new GenesisException("List is empty, nothing to show");
+        }
```

Comment:
This is good! However, exception handling is not consistent across the methods in this class.
Please be consistent and do error handling in all methods.